addons/stock_landed_costs/models/stock_landed_cost.py

```diff
@@ -185,12 +185,13 @@ def button_validate(self):
 
             # batch standard price computation avoid recompute quantity_svl at each iteration
             products = self.env['product.product'].browse(p.id for p in cost_to_add_byproduct.keys()).with_company(cost.company_id)
+            precision_digits = self.env['decimal.precision'].precision_get('Product Unit of Measure')
```

Comment:
You can move it outside of the loop. Unless github is missleading, you're still in the `for cost in self:` loop.